src/components/Post/Post.js

```diff
@@ -24,10 +24,28 @@ const Post = ({ post }) => {
   const { tags, title, description, date, minutes } = post.frontmatter;
   const { author } = useSiteMetadata();
 
+  let timeout;
+  const scrollRef = useRef();
+
   const scrollToTop = () => {
     window.scrollTo({ top: 0, behavior: 'smooth' });
   };
 
+  useEffect(() => {
+    const onScroll = () => {
+      scrollRef.current.style.opacity = 0;
+      clearTimeout(timeout);
+
+      timeout = setTimeout(() => {
+        scrollRef.current.style.opacity = 1;
```

Comment:
another way is to have an `isScrolling` state variable, and set the css of the div inline or something? then don't need `useRef` what do you think

Reply:
like this:

```jsx
const [opacity, setOpacity] = useState(1);
let timeoutID;

useEffect(() => {
  const onScroll = () => {
    setOpacity(0);
    clearTimeout(timeoutID);

    timeoutID = setTimeout(() => {
      setOpacity(1);
    }, 100);
  };

  window.addEventListener('scroll', onScroll);

  return () => {
    window.removeEventListener('scroll', onScroll);
    clearTimeout(timeoutID);
  };
}, []);

<div onClick={scrollToTop} style={{ opacity }} />;
```